dg.go

```diff
@@ -438,6 +447,9 @@ func (n *node[NodeType]) dependencyResolved(dependencyNodeID string, dependencyR
 			panic(ErrConnectionDoesNotExist{dependencyNodeID, n.id})
 		}
 	}
+	if dependencyResolution == Resolved {
+		n.resolvedDependencies[dependencyNodeID] = dependencyType
```

Comment:
If you're locking when retrieving the map in `ResolvedDependencies`, don't you want to lock for consistency when *changing* the map?

Reply:
The lock is supposed to be acquired for the entirety of that function's scope. So it's locked there.

Reply:
OK, I'll trust you rather than searching up the call tree. 😆

Reply:
You don't have to search the call tree -- just look at the function header comment:  it says you have to have the lock when you call the function, so I'm sure we're covered.  😉